video/maintenance.go

```diff
@@ -0,0 +1,122 @@
+package video
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/c2h5oh/datasize"
+	"github.com/lbryio/transcoder/formats"
+	"github.com/lbryio/transcoder/queue"
+)
+
+func toGB(s uint64) string {
+	return fmt.Sprintf("%.2fGB", datasize.ByteSize(s).GBytes())
+}
+
+func SpawnLibraryCleaning(lib *Library) chan<- bool {
+	mls := toGB(lib.maxLocalSize)
+	mrs := toGB(lib.maxRemoteSize)
+	logger.Infow(
+		"starting library maintenance",
+		"max_local_size", mls,
+		"max_remote_size", mrs,
+	)
+	furloughTicker := time.NewTicker(5 * time.Minute)
+	retireTicker := time.NewTicker(24 * time.Hour)
+	stopChan := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-furloughTicker.C:
+				if lib.maxLocalSize == 0 {
+					continue
+				}
```

Comment:
fyi instead of checking this every time, you can use a nil channel. since reads from a nil channel block forever, this case will never be entered.

i think in this case the way you did it is probably cleaner that what i'm suggesting (id have to see the nil channel way to be sure). just something to be aware of in case you didn't know this move